apps/src/schoolInfo/hooks/useSchoolInfo.tsx

```diff
@@ -68,18 +68,60 @@ export function useSchoolInfo(initialState: SchoolInfoInitialState) {
     [initialState.schoolName]
   );
 
+  const [state, setState] = useState<{
+    country: string;
+    schoolId: string;
+    schoolZip: string;
+    schoolName: string;
+    schoolsList: SchoolDropdownOption[];
+  }>({
+    country: detectedCountry,
+    schoolId: detectedSchoolId,
+    schoolZip: detectedZip,
+    schoolName: detectedSchoolName,
+    schoolsList: [],
+  });
+
```

Comment:
I decided to make this a state object rather than use multiple useStates. not a major change, but since we're resetting state sometimes, this will be a little cleaner and reduce re-renders

Reply:
Is the one advantage to this that you can reset the entire state at once? Or are there more? Just haven't seen this pattern before so trying to log it in my memory!

Reply:
yeah... I'm not sure it actually makes much of a difference tbh. I was wrangling with some tests and thought this would help, but the issue was in the test itself 🥴 

there are tradeoffs between using a state object vs individual useStates. for any kind of batch update, like `reset` it's better to have the object, but the downside is the possibility of passing stale `prevState`, particularly with async state updates. (as I'm writing this, I'm realizing the schoolsList is set asynchronously... well crap I might want to change this back or split out schoolsList 🤔 )

side note: if you wrap multiple setState calls in a single `act`, react will batch the updates and you can't guarantee they'll be run in the order you wrote them 🤯 

which is why I needed to do this:
![Screenshot 2024-10-31 at 9 48 39 AM](https://github.com/user-attachments/assets/71305e3a-abbb-413b-9d4b-f70c9c439ad4)

Reply:
fyi, I split out the schoolsList state 🤷